File: frontend/ride_app/src/app/model/driver.ts
export interface Driver {
    id: number;  // Equivalent to Long in Java
    user: User;  // This can be another interface or type for User
    vehicleType: string;
    status: 'AVAILABLE' | 'ON_RIDE' | 'UNAVAILABLE';  // Enums for predefined statuses
    location: string;
    rating: number;
    createdAt: string;  // LocalDateTime will be a string in ISO format
    updatedAt: string;  // LocalDateTime will be a string in ISO format
  }
  
  // User Interface (assuming structure based on the code in the Java class)
  export interface User {
    id: number;
    name: string;
    email: string;
    phoneNumber: string;
    // Add any other properties of the User class if necessary
  }
  

File: microservices/ride-service/src/main/java/ride_service/model/Driver.java
package ride_service.model;


import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity
@Data
public class Driver {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

//    @ManyToOne
//    @JoinColumn(name = "user_id", nullable = false)
    private String userId;

    private String vehicleType;
    private String status; // AVAILABLE, ON_RIDE, UNAVAILABLE
    private String location;
    private Double rating;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;


    public Driver(long l, String suv, String available, String s, double v) {
        this.id = l;
        this.vehicleType = suv;
        this.status = available;
        this.location = s;
        this.rating = v;
    }
}
